streams/src/main/java/org/apache/kafka/streams/state/internals/StoreChangeLogger.java

```diff
@@ -56,7 +57,13 @@ void logChange(final K key, final V value) {
         if (collector != null) {
             final Serializer<K> keySerializer = serialization.keySerializer();
             final Serializer<V> valueSerializer = serialization.valueSerializer();
-            collector.send(this.topic, key, value, this.partition, context.timestamp(), keySerializer, valueSerializer);
+            try {
+                collector.send(this.topic, key, value, this.partition, context.timestamp(), keySerializer, valueSerializer);
+            } catch (final StreamsException e) {
```

Comment:
Because this upper layer can provide a more meaningful exception with more context. For example, in this level we can say that we are writing to a store (unlike the SinkNode exception).